cpu/xmc1000/periph/uart.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2015 Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_xmc1000
+ * @{
+ *
+ * @file
+ * @brief       Low-level UART driver implementation
+ *
+ * @author      Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * @}
+ */
+
+#include "periph/gating.h"
+#include "periph/gpio.h"
+#include "periph/uart.h"
+#include "periph_conf.h"
+#include "sched.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* guard file in case no UART device was specified */
+#if UART_NUMOF
+
+/**
+ * @brief Allocate memory to store the callback functions.
+ */
+static uart_isr_ctx_t uart_ctx[UART_NUMOF];
+
+static inline USIC_CH_TypeDef *_usic(uart_t uart)
```

Comment:
No blocker, but `inline` defined functions will most certainly not be inlined compiled with `-Os`.